lib/sax/parser_feedback_simulator.js

```diff
@@ -76,13 +89,31 @@ ParserFeedbackSimulator.prototype._ensureTokenizerMode = function (tn) {
 ParserFeedbackSimulator.prototype._handleStartTagToken = function (token) {
     var tn = token.tagName;
 
-    if (tn === $.SVG)
+    if (tn === $.SVG) {
         this._enterNamespace(NS.SVG);
+        foreignContent.adjustTokenSVGAttrs(token);
+        foreignContent.adjustTokenXMLAttrs(token);
+    }
 
     else if (tn === $.MATH)
         this._enterNamespace(NS.MATHML);
 
+    else if (tn === $.IMAGE)
+        token.tagName = $.IMG;
```

Comment:
I wonder if we really should do this and new lines adjustment for `<textarea>` and stuff? I thought we should be close to the source code as much as possible, so maybe we should just ignore those tests? However, it's debatable. What do you think?